tests/datasource/data_connector/test_configured_asset_azure_data_connector.py

```diff
@@ -0,0 +1,1346 @@
+from unittest import mock
+
+import pytest
+from ruamel.yaml import YAML
+
+import great_expectations.exceptions as ge_exceptions
+from great_expectations import DataContext
+from great_expectations.core import IDDict
+from great_expectations.core.batch import (
+    BatchDefinition,
+    BatchRequest,
+    BatchRequestBase,
+)
+from great_expectations.data_context.util import instantiate_class_from_config
+from great_expectations.datasource.data_connector import (
+    ConfiguredAssetAzureDataConnector,
+)
+
+yaml = YAML()
+
```

Comment:
This test file contains a LOT of tests (along with boilerplate, mocks, and fixtures). Please note that most of these tests are ported over from S3 and utilize the same logic. There are also a few Azure tests for things like auth.

Some key details to be aware of:
* We do not have an equivalent to `moto3` so we utilize `unittest.mock`
  * The behavior here should be straightforward (with explanations provided for more complex behavior)
* Many of the S3 tests combined a few different assertions and situations together; in the interest of keeping these tests isolated, quick, and in adherence with unit-testing philosophies, I've split many of them up into separate tests. Code duplication should be covered by fixtures.

I'm of the opinion that these tests do a solid job of testing aspects of the class but don't give us 100% coverage. Due to the nature of mocks, we abstract away some function calls that could make or break particular tests. If you see any behavior you think should be applied here, please let me know!

In conjunction with integration/E2E tests, I believe this class to be sufficiently covered through our test suite.